ark/phenotyping/som_utils.py

```diff
@@ -1014,64 +1335,84 @@ def cell_consensus_cluster(base_dir, max_k=20, cap=3, column_prefix='cluster',
+    cell_meta_cluster_avgs_and_counts.to_csv(
+        os.path.join(base_dir, cell_meta_cluster_avgs_name),
+        index=False
+    )
 
-    Args:
-        base_dir (str):
-            The path to the data directories
-        cluster_name (str):
-            The name of the file containing the cluster data
-            Created by cluster_cells or cell_consensus_cluster depending on use case
-        column_prefix (str):
-            The prefix of the columns to subset, should be 'cluster' or 'hCluster_cap'
-        cell_cluster_col (str):
-            Name of the column to group values by, should be 'cluster' or 'hCluster_cap'
-        dpi (float):
-            The resolution of the image to save, ignored if save_dir is None
-        center_val (float):
-            value at which to center the heatmap
-        overlay_values (bool):
-            whether to overlay the raw heatmap values on top
-        colormap (str):
-            color scheme for visualization
-        save_dir (str):
-            If specified, a directory where we will save the plot
-        save_file (str):
-            If save_dir specified, specify a file name you wish to save to.
-            Ignored if save_dir is None
-    """
+    # read in the clust_to_meta_name file
+    print(
+        "Mapping meta cluster values onto average number of pixel SOM/meta cluster counts"
+        "across cell SOM clusters"
+    )
+    som_to_meta_data = feather.read_dataframe(
+        os.path.join(base_dir, clust_to_meta_name)
+    ).astype(np.int64)
 
-    # verify the column prefix provided is valid
-    misc_utils.verify_in_list(
-        provided_column_prefix=column_prefix,
-        valid_column_prefixes=['cluster', 'hCluster_cap']
+    # read in the average number of pixel/SOM clusters across all cell SOM clusters
+    cell_som_cluster_avgs_and_counts = pd.read_csv(cluster_avg_path)
+
+    # merge metacluster assignments in
+    cell_som_cluster_avgs_and_counts = pd.merge_asof(
+        cell_som_cluster_avgs_and_counts, som_to_meta_data, on='cell_som_cluster'
     )
 
-    # verify the cell_cluster_col provided is valid
-    misc_utils.verify_in_list(
-        provided_cluster_col=cell_cluster_col,
-        valid_cluster_cols=['cluster', 'hCluster_cap']
+    # resave average number of pixel/SOM clusters across all cell SOM clusters
+    # with metacluster assignments
+    cell_som_cluster_avgs_and_counts.to_csv(
+        cluster_avg_path,
+        index=False
     )
 
-    # average the columns across the cluster column
-    cluster_avgs = compute_cell_cluster_count_avg(os.path.join(base_dir, cluster_name),
-                                                  column_prefix, cell_cluster_col)
+    # compute the weighted channel average expression per cell SOM cluster
+    print("Compute average weighted channel expression across cell SOM clusters")
+    cell_som_cluster_channel_avg = compute_cell_cluster_channel_avg(
```

Comment:
Your comment about the z-scoring made me this of this as well. Zak's tool needs un-zscored data, so this function shouldn't z-score.